neard/tests/node_cluster.rs

```diff
@@ -0,0 +1,106 @@
+use futures::future;
+
+use near_actix_test_utils::{run_actix_until_stop, spawn_interruptible as spawn};
+use near_client::{ClientActor, ViewClientActor};
+use near_primitives::types::{BlockHeight, BlockHeightDelta, NumSeats, NumShards};
+use testlib::{start_nodes, test_helpers::heavy_test};
+
+pub enum ClusterConfigVariant {
+    HeavyTest(bool),
+    Shards(NumShards),
+    ValidatorSeats(NumSeats),
+    LightClients(usize),
+    EpochLength(BlockHeightDelta),
+    GenesisHeight(BlockHeight),
+}
+
+use ClusterConfigVariant::*;
+
+#[derive(Default, Debug)]
+pub struct NodeCluster {
+    dirs: Vec<tempfile::TempDir>,
+    is_heavy: bool,
+    num_shards: Option<NumShards>,
+    num_validator_seats: Option<NumSeats>,
+    num_lightclient: Option<usize>,
+    epoch_length: Option<BlockHeightDelta>,
+    genesis_height: Option<BlockHeight>,
+}
+
+impl NodeCluster {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn with(mut self, config: ClusterConfigVariant) -> Self {
+        match config {
+            HeavyTest(is_heavy) => self.is_heavy = is_heavy,
+            Shards(n) => self.num_shards = Some(n),
+            ValidatorSeats(n) => self.num_validator_seats = Some(n),
+            LightClients(n) => self.num_lightclient = Some(n),
+            EpochLength(l) => self.epoch_length = Some(l),
+            GenesisHeight(h) => self.genesis_height = Some(h),
+        };
+        self
+    }
+
+    pub fn mkdirs_with<F: Fn(usize) -> String>(mut self, capacity: usize, gen_dirname: F) -> Self {
+        self.dirs = Vec::with_capacity(capacity);
+        self.dirs.extend(
+            (0..capacity).map(|index| {
+                tempfile::Builder::new().prefix(&gen_dirname(index)).tempdir().unwrap()
+            }),
+        );
```

Comment:
```suggestion
        self.dirs = 
            (0..capacity).map(|index| {
                tempfile::Builder::new().prefix(&gen_dirname(index)).tempdir().unwrap()
            }).collect();
```

Reply:
Oh right, `Vec` get's the `size_hint` through `Map`'s proxy on `Range`, so it's reserved to capacity.